rfcs/XXXX-gene-sets/text.md

```diff
@@ -0,0 +1,101 @@
+- [US5] If the gene names match those used in the dataset the user should be able to use general cellxgene functionality (color by genes, calculate differential expression levels, etc.) on those genes
+- [US6] Genesets uploaded for one dataset should be available in other datasets?
+
+### If there is a standard format for differential expression output, we should follow that. If not, the format should be as straightforward as possible while serving these above cases.
+
+### [Optional] Nonfunctional requirements
+- Users should be able to export precomputed differential expression results in real time
+- Users should be able to upload gene sets and have them immediately appear in the right side bar
+
+## Detailed Design | Architecture | Implementation
+
+
+### [Optional] Data model
+
+Fully supporting persistence of differential expression results is out of the scope of this RFC. But the data model described below has been designed to allow for differential expression persistence in the future.
+###Relational Database
+To support gene sets four tables will be added to the cellxgene relational database
+- Gene
+   - UUID (generated)
+   - GeneName
+   - OntologyId?
+- GeneGeneSetLink
+   - UUID
+   - Gene UUID
+   - GeneSet UUID
+   - Comments (optional)
+- GeneSet
+   - UUID (generated)
+   - GeneSet Name (required)
+   - Gene Count (generated)
+   - Comments (optional, only available when the gene set is generated by cellxgene)
+- GeneSetUserLink
+   - User UUID
+   - GeneSet UUID
+   - Comments (optional)
+   - Dataset UUID (?)
+
+![Cellxgene Data Schema](imgs/Cellxgene_rds_schema.png)
+**Figure 1** Cellxgene data schema, tables to be added (as described above) are in red
+#### Imported CSV
+GeneSet CSV
+- A file containing a comma separated list of the gene set name, genes and (optionally) details about why they were included. Each gene should be on a new line.
+```
+GENESET,GENE,COMMENTS
+SET1, a23,I picked this gene because I think it causes cancer
+SET1, b46,I picked this gene because I think it prevents cancer
+SET2, c19,I picked this gene because it has a funny name
+SET2, d57,
+```
+#### Exported CSV
+Exported File
+-  A file containing a comma separated list of genes, logfold values and p-values. The name of the directory storing the file will reference the dataset and the name of the file will contain names of the cell sets being compared, see datasetID/categoryA.label1-categoryB.label2.csv for an examole
+In the future, if cells from two different datasets are being compared the dataset id will be appended as a prefix to the category names in the filename (?will this make the filename too long?)
+```
+GENE,PVALUE,LOGFOLD
+a23,0.03,3
+b46,0.02,2
+c19,0.05,1
+```
+
+## [Optional] Alternatives
+
+One simplifying alternative is to have the format of the exported and the imported csvs match. 
+However this doesnt match described use cases or user behavior. Logfold and P-values were left out of the imported csv 
```

Comment:
What is the value in having different formats for import and export? I think this could be confusing for a user and does not actually fit the use case. A user would want to import DE results they run in scVI for example and include the test statistic

Reply:
Discussed in person